### PR TITLE
Use API calls in load_demo_data whenever possible

### PR DESCRIPTION
**Pull request by acrellin**
Closes https://github.com/skyportal/skyportal/issues/246

### CODE REVIEW

File: tools/load_demo_data.py [hunk 1]
```diff
@@ -34,80 +37,191 @@
...
-                              public_url=f'/static/thumbnails/{fname}')
-                DBSession().add(t)
-                shutil.copy(basedir/f'skyportal/tests/data/{fname}', basedir/'static/thumbnails/')
-
-            s.add_linked_thumbnails()
+    with status("Launching web app & executing API calls"):
+        web_client = subprocess.Popen(['make', 'run'],
+                                      cwd=basedir, preexec_fn=os.setsid)
+
+        server_url = f"http://localhost:{cfg['ports.app']}"
```
[the remark this answers is not in the document]
Comment (acrellin) [reply tools/load_demo_data.py]: @stefanv this is consistent with every other instance of URL construction in our app:  `f'http://localhost:{cfg["ports.app"]}'`

**Comment by acrellin** [comment]
@stefanv thanks for the feedback - I've addressed all of your comments/suggestions!

File: tools/load_demo_data.py [hunk 2]
```diff
@@ -34,80 +38,168 @@
...
+
+                data = assert_post('instrument',
+                                   data={'name': 'P60 Camera',
+                                         'type': 'phot',
+                                         'band': 'optical',
+                                         'telescope_id': telescope1_id})
+                instrument1_id = data['data']['id']
+
+                data = assert_post('telescope',
+                                            data={'name': 'Nordic Optical Telescope',
```
Comment (stefanv) [inline tools/load_demo_data.py]: Indentation

File: tools/load_demo_data.py [hunk 3]
```diff
@@ -34,80 +38,168 @@
...
+
+                    data = assert_post('sources', data=source_info)
+                    assert data['data']['id'] == source_info['id']
+
+                    for comment in comments:
+                        data = assert_post('comment',
+                                           data={'source_id': source_info['id'],
+                                                 'text': comment})
+
+                    phot_file = os.path.join(os.path.dirname(os.path.dirname(__file__)),
```
Comment (stefanv) [inline tools/load_demo_data.py]: I know you enjoy using `pathlib.Path` — maybe that would look nice here.  Your call.  Can also just be `skyportal/tests/data/phot.csv`.

File: tools/load_demo_data.py [hunk 4]
```diff
@@ -34,80 +38,168 @@
...
+                            open(os.path.abspath(fpath), 'rb').read())
+                        data = assert_post('thumbnail',
+                                           data={'source_id': source_info['id'],
+                                                 'data': thumbnail_data,
+                                                 'ttype': ttype
+                                           })
+
+                    source = Source.query.get(source_info['id'])
+                    source.add_linked_thumbnails()
+        except Exception as e:
```
Comment (stefanv) [inline tools/load_demo_data.py]: I'd remove this except clause, since it doesn't seem to do anything useful.

File: tools/load_demo_data.py [hunk 5]
```diff
@@ -34,80 +38,168 @@
...
+                                           data={'source_id': source_info['id'],
+                                                 'text': comment})
+
+                    phot_file = os.path.join(os.path.dirname(os.path.dirname(__file__)),
+                                             'skyportal', 'tests', 'data',
+                                             'phot.csv')
+                    phot_data = pd.read_csv(phot_file)
+
+                    data = assert_post('photometry',
+                                       data={'source_id': source_info['id'],
```
Comment (stefanv) [inline tools/load_demo_data.py]: I'm actually quite curious what [black](https://pypi.org/project/black/) would recommend in this case.

**Pull request merged**